lib/include/TransportDescriptor.hpp

```diff
@@ -0,0 +1,2033 @@
+void clear_enable_tcp_nodelay(
+        const std::string& xml_file,
+        const std::string& transport_descriptor_id);
+
+/**
+ * @brief Remove shared memory segment size from SHM Transport Descriptor profile.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ *
+ * @throw file_not_found Exception if the provided XML file is not found/readable.
+ * @throw element_not_found Exception if the specified transport descriptor profile is not found in the XML file.
+ */
+void clear_segment_size(
+        const std::string& xml_file,
+        const std::string& transport_descriptor_id);
+
+/**
+ * @brief Remove shared memory listening port capacity from SHM Transport Descriptor profile.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ *
+ * @throw file_not_found Exception if the provided XML file is not found/readable.
+ * @throw element_not_found Exception if the specified transport descriptor profile is not found in the XML file.
+ */
+void clear_port_queue_capacity(
+        const std::string& xml_file,
+        const std::string& transport_descriptor_id);
+
+/**
+ * @brief Remove shared memory listening port liveliness timeout from SHM Transport Descriptor profile.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ *
+ * @throw file_not_found Exception if the provided XML file is not found/readable.
+ * @throw element_not_found Exception if the specified transport descriptor profile is not found in the XML file.
+ */
+void clear_healthy_check_timeout_ms(
+        const std::string& xml_file,
+        const std::string& transport_descriptor_id);
+
+/**
+ * @brief Remove shared memory debugging file path from SHM Transport Descriptor profile.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ *
+ * @throw file_not_found Exception if the provided XML file is not found/readable.
+ * @throw element_not_found Exception if the specified transport descriptor profile is not found in the XML file.
+ */
+void clear_rtps_dump_file(
+        const std::string& xml_file,
+        const std::string& transport_descriptor_id);
+
+/************************************************************************/
+/* Set functions                                                        */
+/************************************************************************/
+
+/**
+ * @brief Set the Transport Descriptor kind.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ * @param[in] kind Transport descriptor kind.
+ *
+ * @throw element_invalid Exception if the provided kind is not valid.
```

Comment:
I think that all `set`functions should also throw the `file_not_found` and `element_not_found` exceptions.

Reply:
This is not correct, if the file is not found the library should create a new XML file. And if the element is not found, the library should create it with the `set` command. The only case where the `element_not_found` exception should be thrown is in case of updating a list/map element.